state/validation.go

```diff
@@ -137,14 +141,35 @@ func (state *State) ValidatorLoop() {
 				for i := 0; i < 10 && state.InMsgQueue().Length() == 0 && state.InMsgQueue2().Length() == 0 && len(s.Holding) < 5; i++ {
 					time.Sleep(10 * time.Millisecond)
 					state.ValidatorLoopSleepCnt++
+					if shouldShutdown(state) {
+						return
+					}
 				}
 
 			}
 		}
+	}
+}
 
+func shouldShutdown(state *State) bool {
+	select {
+	case <-state.ShutdownChan:
+		shutdown(state)
+		return true
+	default:
+		return false
 	}
 }
 
+func shutdown(state *State) {
+	state.RunState = runstate.Stopping
+	fmt.Println("Closing the Database on", state.GetFactomNodeName())
+	state.DB.Close()
+	state.StateSaverStruct.StopSaving() // Shouldn't this be done before closing the database?
```

Comment:
Shouldn't this be done before closing the database? Or is there a specific reason to do this after?

Reply:
The savestate doesn't pull it's data from the db. So it isn't actually dependent on the db